src/main/java/io/codelex/flightplanner/flights/FlightsDatabaseService.java

```diff
@@ -0,0 +1,106 @@
+
+    private AirportRepository airportRepository;
+
+    public FlightsDatabaseService(FlightsRepository flightsRepository, AirportRepository airportRepository) {
+        this.flightsRepository = flightsRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    @Override
+    @Transactional
+    public Flight addFlight(AddFlightRequest addFlightRequest) throws DepartureAndArrivalAirportAreTheSameException, IllegalArgumentException, FlightAlreadyExistException, NullPointerException {
+        if (addFlightRequest.isAirportFromAndToEqual()) {
+            throw new DepartureAndArrivalAirportAreTheSameException("Departure and arrival airport are  the same!");
+        }
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        if (!addFlightRequest.isCorrectArrivalTime()) {
+            throw new IllegalArgumentException();
+        }
+        if (!this.airportRepository.existsByAirportAndCityAndCountry(addFlightRequest.getFrom().getAirport(),
+                addFlightRequest.getFrom().getCity(),
+                addFlightRequest.getFrom().getCountry())) {
+            Airport from = new Airport();
+            from.setAirport(addFlightRequest.getFrom().getAirport());
+            from.setCity(addFlightRequest.getFrom().getCity());
+            from.setCountry(addFlightRequest.getFrom().getCountry());
+            this.airportRepository.save(from);
+        }
+        if (!this.airportRepository.existsByAirportAndCityAndCountry(addFlightRequest.getTo().getAirport(),
+                addFlightRequest.getTo().getCity(),
+                addFlightRequest.getTo().getCountry())) {
+            Airport to = new Airport();
+            to.setAirport(addFlightRequest.getTo().getAirport());
+            to.setCity(addFlightRequest.getTo().getCity());
+            to.setCountry(addFlightRequest.getTo().getCountry());
+            this.airportRepository.save(to);
+        }
+        if (this.flightsRepository.existsByCarrierAndFromAndToAndDepartureTimeAndArrivalTime(addFlightRequest.getCarrier(),
+                addFlightRequest.getFrom(), addFlightRequest.getTo(), LocalDateTime.parse(addFlightRequest.getDepartureTime(), formatter), LocalDateTime.parse(addFlightRequest.getArrivalTime(), formatter))) {
+            throw new FlightAlreadyExistException("Flight already exists!");
+        }
+        Flight newFlight = new Flight();
+        newFlight.setCarrier(addFlightRequest.getCarrier());
+        newFlight.setFrom(addFlightRequest.getFrom());
+        newFlight.setTo(addFlightRequest.getTo());
+        newFlight.setDepartureTime(LocalDateTime.parse(addFlightRequest.getDepartureTime(), formatter));
+        newFlight.setArrivalTime(LocalDateTime.parse(addFlightRequest.getArrivalTime(), formatter));
+        this.flightsRepository.save(newFlight);
+        return this.flightsRepository.searchFlightByFromAndToAndCarrierAndDepartureTimeAndArrivalTime(addFlightRequest.getFrom(),
```

Comment:
Manuprāt save metode kuru sauc rindņu augstāk jau atgriež saglabāto objektu, attiecīgi meklēšana pēc viņa vēlreiz nav nepieciešama.